swan/scripts/before-notebook.d/02_environment.sh

```diff
@@ -118,7 +130,12 @@ then
   exit 1
 else
   CONFIGURE_KERNEL_ENV_TIME_SEC=$(echo $(date +%s.%N --date="$START_TIME_CONFIGURE_KERNEL_ENV seconds ago") | bc)
-  _log "user: $NB_USER, host: ${SERVER_HOSTNAME%%.*}, metric: configure_kernel_env.${ROOT_LCG_VIEW_NAME:-none}.${SPARK_CLUSTER_NAME:-none}.duration_sec, value: $CONFIGURE_KERNEL_ENV_TIME_SEC"
+  software_source_name="customenv"
```

Comment:
It is clearer and less error prone to always set variable names depending on a condition. I would not initialize `software_source_name` to `customenv` if you have not checked that the software source is a custom env.

I'd remove this line, and below add an `elif [ "$SOFTWARE_SOURCE" == "customenv" ]; then`